### PR TITLE
A few helper methods for common wait functionality.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

A few helper methods for common wait functionality. These are needed in many examples so it is best to avoid the repetition.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/endpoints.rs [hunk 1]
```diff
@@ -883,6 +883,53 @@ impl Client {
...
+    ///
+    /// Since this can take an indefinite amount of time in general, users of
+    /// this function might wish to wrap it inside
+    /// [`timeout`](tokio::time::timeout) handler and handle the resulting
+    /// failure.
+    pub async fn wait_for_finalization(
+        &mut self,
+        height: types::AbsoluteBlockHeight,
+    ) -> RPCResult<(types::AbsoluteBlockHeight, types::hashes::BlockHash)> {
+        let mut interval = tokio::time::interval(tokio::time::Duration::from_millis(250));
```
Comment (limemloh) [inline src/endpoints.rs:897]: I would probably move the polling interval duration up as a parameter.

File: src/endpoints.rs [hunk 2]
```diff
@@ -883,6 +883,53 @@ impl Client {
...
+    ///
+    /// Since this can take an indefinite amount of time in general, users of
+    /// this function might wish to wrap it inside
+    /// [`timeout`](tokio::time::timeout) handler and handle the resulting
+    /// failure.
+    pub async fn wait_for_blocks_at(
+        &mut self,
+        height: types::AbsoluteBlockHeight,
+    ) -> RPCResult<Vec<types::hashes::BlockHash>> {
+        let mut interval = tokio::time::interval(tokio::time::Duration::from_millis(250));
```
Comment (limemloh) [inline src/endpoints.rs:921]: Same here and the rest of the wait functions

File: src/types/basic.rs [hunk 3]
```diff
@@ -694,6 +694,18 @@ pub struct UpdateKeysThreshold {
     pub(crate) threshold: u16,
 }
 
+impl TryFrom<u16> for UpdateKeysThreshold {
+    type Error = &'static str;
```
Comment (limemloh) [inline src/types/basic.rs:698]: I would prefer a proper error type here.

**Review by limemloh — APPROVED** [review]
Looks good only some considerations.

File: src/endpoints.rs (hunk 2 above)
Comment (abizjak) [reply src/endpoints.rs:921]: See my comment above.

File: src/types/basic.rs (hunk 3 above)
Comment (abizjak) [reply src/types/basic.rs:698]: Yes. It's just another unit type though :)

But will add.

Comment (abizjak) [reply src/types/basic.rs]: Added.

**Pull request merged**